app/controllers/api/v1/materials_controller.rb

```diff
@@ -29,11 +29,33 @@ def create
 
   # PATCH/PUT /materials/1
   def update
-    if @material.update(material_params)
-      render json: @material
-    else
+    material_type = @material.material_type
+    unless material_type_update_params[:relationships].nil? or material_type_update_params[:relationships][:material_type].nil?
+      material_type = MaterialType.find_by(material_type_update_params[:relationships][:material_type][:data][:attributes])
+    end
+
+    begin
+      ActiveRecord::Base.transaction do
```

Comment:
Seemed a bit strange to me to use a transaction here, so I was reading a little about it (mainly in http://stackoverflow.com/questions/15656858/rails-3-transactions-in-controllers). They say that transactions should not be used inside the controller, it is considered an anti-pattern. Maybe we could move it to some other place in the model?